src/domain/file/services/file-service.ts

```diff
@@ -109,16 +110,61 @@ export class FileService {
         return await this.fileRepository.getSupportingFilesBySubmissionId(submissionId);
     }
 
-    async upload(fileContents: Buffer, file: File): Promise<S3.ManagedUpload.SendData> {
+    async uploadSupportManuscript(
+        fileContents: Buffer,
+        file: File,
+        userId: string,
+        pubsub: PubSub,
+    ): Promise<S3.ManagedUpload.SendData> {
         const { url, id, mimeType } = file;
-        return this.s3
-            .upload({
-                Bucket: this.bucket,
-                Key: `${url}/${id}`,
-                Body: fileContents.toString(),
-                ContentType: mimeType,
-                ACL: 'private',
-            })
-            .promise();
+        const fileUploadManager = this.s3.upload({
+            Bucket: this.bucket,
+            Key: `${url}/${id}`,
+            Body: fileContents.toString(),
+            ContentType: mimeType,
+            ACL: 'private',
+        });
+
+        fileUploadManager.on('httpUploadProgress', async ({ loaded, total }) => {
+            await pubsub.publish('UPLOAD_STATUS', {
+                manuscriptUploadProgress: {
+                    userId,
+                    filename: file.filename,
+                    fileId: file.id,
+                    percentage: Math.floor((loaded / total) * 100),
+                },
+            });
+        });
+
+        return fileUploadManager.promise();
+    }
+
+    async uploadSupportingFile(
+        fileContents: Buffer,
+        file: File,
+        userId: string,
+        pubsub: PubSub,
+    ): Promise<S3.ManagedUpload.SendData> {
+        const { url, id, mimeType } = file;
+        const fileUploadManager = this.s3.upload({
+            Bucket: this.bucket,
+            Key: `${url}/${id}`,
+            Body: fileContents.toString(),
+            ContentType: mimeType,
+            ACL: 'private',
+        });
+
+        fileUploadManager.on('httpUploadProgress', async ({ loaded, total }) => {
+            await pubsub.publish('UPLOAD_STATUS', {
+                supportingUploadProgress: {
+                    userId,
+                    filename: file.filename,
+                    fileId: file.id,
+                    percentage: Math.floor((loaded / total) * 100),
```

Comment:
Happy to go with this for now - but xPub does different. Comments?

Reply:
One for @erezmus - I'm not very aware of xpub.

Reply:
See https://github.com/elifesciences/elife-xpub/blob/develop/packages/component-submission/server/use-cases/helpers/upload-predictor.js

Reply:
I believe we're doing away with predictions now for realistic % upload and indeterminate processing state once the files stored